docs/csharp/language-reference/compiler-messages/cs8410.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Compiler Error CS8410
+ms.date: 07/11/2020
+f1_keywords:
+  - "CS8410"
+helpviewer_keywords:
+  - "CS8410"
+author: Youssef1313
+---
+# Compiler Error CS8410
+
+'type': type used in an asynchronous using statement must be implicitly convertible to 'System.IAsyncDisposable' or implement a suitable 'DisposeAsync' method.
+
+The expression inside an `await using` statement must have a `DisposeAsync` method.
+
+## To correct this error
+
+Use `using` instead of `await using`.
+
+## Example
+
+```csharp
+using System.Threading.Tasks;
+
+static class Program
+{
+    static async Task Main()
+    {
+        // error CS8410: 'Example': type used in an asynchronous using statement
+        // must be implicitly convertible to 'System.IAsyncDisposable' or implement
+        // a suitable 'DisposeAsync' method.
+        await using var reader = new Example();
+    }
+}
+
+class Example
+{
+}
+```
```

Comment:
```suggestion
```csharp
using System.Threading.Tasks;

static class Program
{
    static async Task Main()
    {
        // error CS8410: 'Example': type used in an asynchronous using statement
        // must be implicitly convertible to 'System.IAsyncDisposable' or implement
        // a suitable 'DisposeAsync' method.
        await using var example = new Example();
    }
}

class Example
{
}
```
```